python/graphstorm/inference/emb_infer.py

```diff
@@ -0,0 +1,177 @@
+        infer_ntypes = set()
+        for etype in g.infer_idxs:
+            infer_ntypes.add(etype[0])
+            infer_ntypes.add(etype[2])
+        infer_ntypes = sorted(infer_ntypes)
+
+        if use_mini_batch_infer:
+            embs = do_mini_batch_inference(model, g, fanout=fanout,
+                                           edge_mask=None,
+                                           task_tracker=self.task_tracker,
+                                           infer_ntypes=infer_ntypes)
+        else:
+            embs = do_full_graph_inference(model, g, fanout=fanout,
+                                           task_tracker=self.task_tracker)
+        return embs
+
+    def lp_emb(self, g, model, use_mini_batch_infer, fanout):
+        """ Embedding Generation for link prediction task.
+        It will only generate embeddings on whole graph.
+
+        Parameters
+        ----------
+        g: GSgnnData
+            The GraphStorm dataset
+        model : GSgnnNodeModel
+            The GNN model on edge prediction/classification task
+        use_mini_batch_infer : bool
+            Whether to use mini-batch inference when computing node embeddings.
+        fanout: list of int
+            The fanout of each GNN layers used in inference.
+        """
+        if use_mini_batch_infer:
+            embs = do_mini_batch_inference(model, g, fanout=fanout,
+                                           edge_mask=None,
+                                           task_tracker=self.task_tracker)
+        else:
+            embs = do_full_graph_inference(model, g, fanout=fanout,
+                                           edge_mask=None,
+                                           task_tracker=self.task_tracker)
+        return embs
+
+    def infer(self, data, task_type, save_embed_path, eval_fanout,
+            use_mini_batch_infer=False,
+            node_id_mapping_file=None,
+            save_embed_format="pytorch"):
+        """ Do Embedding Generating
+
+        Generate node embeddings and save into disk.
+
+        Parameters
+        ----------
+        data: GSgnnData
+            The GraphStorm dataset
+        task_type : str
+            task_type must be one of graphstorm builtin task types
+        save_embed_path : str
+            The path where the GNN embeddings will be saved.
+        use_mini_batch_infer : bool
+            Whether to use mini-batch inference when computing node embeddings.
+        node_id_mapping_file: str
+            Path to the file storing node id mapping generated by the
+            graph partition algorithm.
+        save_embed_format : str
+            Specify the format of saved embeddings.
+        """
+
+        device = self.device
+
+        assert save_embed_path is not None, \
+            "It requires save embed path for gs_gen_node_embedding"
+
+        sys_tracker.check('start embedding generation')
+        self._model.eval()
+
+        if task_type == BUILTIN_TASK_LINK_PREDICTION:
```

Comment:
Code is straightforward here. We just need to do:
for different task types, we get ntypes to do inference
then do inference.

So we do not need to implement lp_emb, nc_emb, etc.